### PR TITLE
Add ClearAroundRobot and ClearExceptRegion action bt

### PR DESCRIPTION
**Pull request by doisyg**
ClearEntireCostmap action bt was implemented but the 2 others (ClearAroundRobot and ClearExceptRegion) were missing

### CODE REVIEW

File: nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/clear_costmap_service.hpp [hunk 1]
```diff
@@ -19,6 +19,8 @@
 
 #include "nav2_behavior_tree/bt_service_node.hpp"
```
Comment (SteveMacenski) [inline nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/clear_costmap_service.hpp:20]: https://app.circleci.com/pipelines/github/ros-planning/navigation2/4713/workflows/45fd3136-d3a2-47d7-b7f4-1f161b1fbef1/jobs/17524 some linting errors, you can run these locally with `ament_cpplint` and `ament_uncrustify` so you can see these before CI runs

File: nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/clear_costmap_service.hpp [hunk 2]
```diff
@@ -33,6 +35,44 @@ class ClearEntireCostmapService : public BtServiceNode<nav2_msgs::srv::ClearEnti
...
+    const std::string & service_node_name,
+    const BT::NodeConfiguration & conf);
+
+  void on_tick() override;
+
+  static BT::PortsList providedPorts()
+  {
+    return providedBasicPorts(
+      {
+        BT::InputPort<int>("reset_distance", 1,
```
Comment (SteveMacenski) [inline nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/clear_costmap_service.hpp:51]: Shouldn't this be a double?

File: nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/clear_costmap_service.hpp [hunk 3]
```diff
@@ -33,6 +35,44 @@ class ClearEntireCostmapService : public BtServiceNode<nav2_msgs::srv::ClearEnti
...
```
Comment (SteveMacenski) [inline nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/clear_costmap_service.hpp:70]: Shouldn't this be a double?

File: nav2_behavior_tree/plugins/action/clear_costmap_service.cpp [hunk 4]
```diff
@@ -32,10 +32,38 @@ void ClearEntireCostmapService::on_tick()
   increment_recovery_count();
 }
 
+ClearExceptRegionService::ClearExceptRegionService(
```
Comment (SteveMacenski) [inline nav2_behavior_tree/plugins/action/clear_costmap_service.cpp:35]: https://github.com/ros-planning/navigation2/blob/80bea85151abe264a3f9297694f0f863f7f7dc05/nav2_behavior_tree/test/plugins/action/test_clear_costmap_service.cpp if you added simple test fixtures for these, that would give them test coverage

**Review by SteveMacenski — COMMENTED** [review]
LGTM, just needs some additional stuff to make it complete
- Some simple tests (analogs in comment) 
- https://github.com/ros-planning/navigation2/tree/main/nav2_behavior_tree the readme needs to have these added
- https://navigation.ros.org/configuration/packages/configuring-bt-xml.html added to the BT XML list
- https://navigation.ros.org/plugins/index.html added to the full navigation plugin list 
- https://navigation.ros.org/migration/Foxy.html and migration guide

File: nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/clear_costmap_service.hpp (hunk 1 above)
Comment (doisyg) [reply nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/clear_costmap_service.hpp:20]: Thx for the tips

**Comment by doisyg** [comment]
Also renamed the added bt action names for consistency with the srv names

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
Just needs doc updates on the website: https://github.com/ros-planning/navigation.ros.org to merge!

**Comment by doisyg** [comment]
> Just needs doc updates on the website: https://github.com/ros-planning/navigation.ros.org to merge!

Hum, looks like this will require more that direct github edit this time. Any advises on a to proceed efficiently ? Local fork of https://github.com/ros-planning/navigation.ros.org/ + regular PR ? Any tips on how to preview locally the read the docs pages ?

**Comment by SteveMacenski** [comment]
> Hum, looks like this will require more that direct github edit this time. Any advises on a to proceed efficiently ? Local fork of https://github.com/ros-planning/navigation.ros.org/ + regular PR ? Any tips on how to preview locally the read the docs pages ?

... yes probably need to make a branch to PR against :laughing: . `make html` will build locally and you can go into `_build` to find the `index.html` page to view the local version to verify. There's some dependencies you need to install to build I don't have a guide for, but CI does it https://github.com/ros-planning/navigation.ros.org/blob/master/.circleci/config.yml so check there for anything missing.

**Comment by doisyg** [comment]
Readme done, I will attempt the rst hopefully soon, thx for the guidelines

File: nav2_behavior_tree/README.md [hunk 5]
```diff
@@ -78,7 +78,10 @@ The nav2_behavior_tree package provides several navigation-specific nodes that a
...
-| Spin | Action | Invokes the Spin ROS2 action server, which is implemented by the nav2_recoveries module. This action is using in nav2 Behavior Trees as a recovery behavior. |
+| Spin | Action | Invokes the Spin ROS2 action server, which is implemented by the nav2_recoveries module. This action is used in nav2 Behavior Trees as a recovery behavior. |
 | PipelineSequence | Control | Ticks the first child till it succeeds, then ticks the first and second children till the second one succeeds. It then ticks the first, second, and third children until the third succeeds, and so on, and so on. If at any time a child returns RUNNING, that doesn't change the behavior. If at any time a child returns FAILURE, that stops all children and returns FAILURE overall.|
+| ClearEntireCostmap | Action | Invokes the ClearEntireCostmap ROS2 service server of costmap_2d_ros, which is implemented by the nav2_recoveries module. The service address needs to be specified using the `service_name` input port, typically `local_costmap/clear_entirely_local_costmap` or `global_costmap/clear_entirely_global_costmap`. This action is used in nav2 Behavior Trees as a recovery behavior. |
```
Comment (SteveMacenski) [inline nav2_behavior_tree/README.md:83]: This has no relationship with nav2_recoveries, it goes directly to the costmaps.

Comment (SteveMacenski) [reply nav2_behavior_tree/README.md:83]: As well for both below

File: nav2_behavior_tree/README.md [hunk 6]
```diff
@@ -78,7 +78,10 @@ The nav2_behavior_tree package provides several navigation-specific nodes that a
...
+| ClearCostmapExceptRegion | Action | Invokes the ClearCostmapExceptRegion ROS2 service server of costmap_2d_ros, which is implemented by the nav2_recoveries module. The service address needs to be specified using the `service_name` input port, typically `local_costmap/clear_except_local_costmap` or `global_costmap/clear_except_global_costmap`. This action is used in nav2 Behavior Trees as a recovery behavior. |
```
Comment (SteveMacenski) [inline nav2_behavior_tree/README.md:84]: Please explain a little about what these bottom 2 do (how do they differ from clear entire?)

**Comment by doisyg** [comment]
Docs here: https://github.com/ros-planning/navigation.ros.org/pull/144

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
Awesome, thanks for the contribution!

**Comment by SteveMacenski** [comment]
@doisyg https://app.circleci.com/pipelines/github/ros-planning/navigation2/4803/workflows/dee88964-7c50-41f1-b9aa-dd1f024fc603/jobs/17814 this PR is making CI crash in `main`

**Comment by doisyg** [comment]
Sorry, fixed here: https://github.com/ros-planning/navigation2/pull/2224

**Pull request merged**